frost/bip340.go

```diff
@@ -52,6 +52,12 @@ func (bc *Bip340Curve) EcAdd(a *Point, b *Point) *Point {
 	return &Point{x, y}
 }
 
+// EcSub returns the subtraction of two elliptic curve points.
+func (bc *Bip340Curve) EcSub(a *Point, b *Point) *Point {
+	bNeg := &Point{b.X, new(big.Int).Neg(b.Y)}
```

Comment:
`EcSub` calls `EcAdd` that calls `BitCurve.Add` which [executes modulo over the order of the underlying field](https://github.com/ethereum/go-ethereum/blob/e3eeb64c9424d599efc3d3f9cde1c64131f694aa/crypto/secp256k1/curve.go#L118-L122) (`BitCurve.P`) on the result. Unless we are talking about doing it for x and y before the `EcAdd` is called? If so, I think we should consider doing it in `DeserializePoint` function before checking if the point is on the curve.

Reply:
It's not a big deal precisely because of that, but given that `EcSub` isn't called that often and doing the negation modularly isn't very expensive, I think it might be somewhat preferable to try to stay within the invariants just for clarity's sake so that it doesn't _look_ like a potential bug (like `bNeg := &Point{b.X, new(big.Int).Sub(bc.FieldOrder(), b.Y)}`), or add a comment explaining why this shortcut works fine.

Reply:
Done in #10 in e16596e5635a28cff33218546d818ee211d98cff.